src/main/scala/chisel3/stage/phases/AddDedupGroupAnnotations.scala

```diff
@@ -0,0 +1,41 @@
+package chisel3.stage.phases
+
+import firrtl.AnnotationSeq
+import firrtl.options.{Dependency, Phase}
+import firrtl.options.Viewer.view
+import firrtl.transforms.DedupGroupAnnotation
+import scala.annotation.nowarn
+
+import chisel3.experimental.BaseModule
+import chisel3.stage._
+import chisel3.stage.CircuitSerializationAnnotation._
+import chisel3.ChiselException
+import chisel3.experimental.dedupGroup
+import chisel3.internal.firrtl.DefBlackBox
+
+class AddDedupGroupAnnotations extends Phase {
+  override def prerequisites = Seq.empty
+  override def optionalPrerequisites = Seq(Dependency[Convert])
+  override def optionalPrerequisiteOf = Seq.empty
+  override def invalidates(a: Phase) = false
+
+  def transform(annotations: AnnotationSeq): AnnotationSeq = {
+    val chiselOptions = view[ChiselOptions](annotations)
+    val circuit = chiselOptions.chiselCircuit.getOrElse {
+      throw new ChiselException(
+        s"Unable to locate the elaborated circuit, did ${classOf[Elaborate].getName} run correctly"
+      )
+    }
+
+    val skipAnnos = annotations.collect { case x: DedupGroupAnnotation => x.target }
```

Comment:
```suggestion
    val skipAnnos = annotations.collect { case x: DedupGroupAnnotation => x.target }.toSet
```
Need a `.toSet` here or the contains below are going to be very slow.